hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/QueuePlacementConverter.java

```diff
@@ -187,7 +188,7 @@ private Rule createNestedRule(Policy policy,
       boolean create,
       FSConfigToCSConfigRuleHandler ruleHandler,
       boolean fsParentCreate,
-      String parentQueue,
+      QueuePath parentQueue,
```

Comment:
Was this really the full queue path before? or just the queue name?

Reply:
As I can see it should be the full queue path. The queue name gets its value from the defaultQueueName member of the DefaultPlacementRule class: https://github.com/apache/hadoop/blob/1f73332bb6632358b8f8272c8c8a8b617a0c6cb0/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/QueuePlacementConverter.java#L134

In the DefaultPlacementRule class we call an assureRoot() method on it before setting it from the configuration, like this: https://github.com/apache/hadoop/blob/1f73332bb6632358b8f8272c8c8a8b617a0c6cb0/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/placement/DefaultPlacementRule.java#L74

If I'm not mistaken with this, it should ensure that we are using the full queue path here as well.